From 3c74f403cb38410ea7e1de0e61dafa80a60c5ba5 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Fri, 10 Jul 2020 09:18:12 +0800
Subject: [PATCH] policy/modules/services/bluetooth: make bluetooth_t domain
 MLS trusted for reading from files up to its clearance

Fixes:
avc:  denied  { search } for  pid=268 comm="bluetoothd" name="journal"
dev="tmpfs" ino=14165
scontext=system_u:system_r:bluetooth_t:s0-s15:c0.c1023
tcontext=system_u:object_r:syslogd_runtime_t:s15:c0.c1023 tclass=dir
permissive=0

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/services/bluetooth.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/services/bluetooth.te b/policy/modules/services/bluetooth.te
index b3df695db..931021346 100644
--- a/policy/modules/services/bluetooth.te
+++ b/policy/modules/services/bluetooth.te
@@ -132,6 +132,8 @@ userdom_dontaudit_search_user_home_dirs(bluetooth_t)
 init_dbus_send_script(bluetooth_t)
 systemd_dbus_chat_hostnamed(bluetooth_t)
 
+mls_file_read_to_clearance(bluetooth_t)
+
 optional_policy(`
 	dbus_system_bus_client(bluetooth_t)
 	dbus_connect_system_bus(bluetooth_t)
-- 
2.17.1

